bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/WorkbenchWindow.java

```diff
@@ -2342,14 +2343,15 @@ public void run(final boolean fork, boolean cancelable, final IRunnableWithProgr
 				final InterruptedException[] ie = new InterruptedException[1];
 
 				BusyIndicator.showWhile(getShell().getDisplay(), () -> {
+					IProgressMonitor progressMonitor = manager.getProgressMonitor();
 					try {
-						ModalContext.run(runnable, fork, manager.getProgressMonitor(), getShell().getDisplay());
+						ModalContext.run(runnable, fork, progressMonitor, getShell().getDisplay());
 					} catch (InvocationTargetException e1) {
 						ite[0] = e1;
 					} catch (InterruptedException e2) {
 						ie[0] = e2;
 					} finally {
-						manager.getProgressMonitor().done();
+						progressMonitor.done();
```

Comment:
This does not fully fix the issue with "Open Implementation"? I still see warning reported.

Reply:
true. interestingly crtl+T works without warning, i have not found the difference yet